dialer.go

```diff
@@ -253,6 +256,11 @@ func NewDialer(ctx context.Context, opts ...Option) (*Dialer, error) {
 	if err != nil {
 		return nil, err
 	}
+	var r InstanceConnectionNameResolver = cloudsql.DefaultInstanceConnectionNameResolver
```

Comment:
What if we used a shorter name, e.g., `DefaultResolver`? We have a public `instance` package where we could put this, the interface, and the SRVResolver.

Reply:
Yes. See review comment.